File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, r: Optional[TreeNode]) -> int:
        if(r==None):
            return 0
        return max(1 + self.maxDepth(r.left), 1 + self.maxDepth(r.right));

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& n) {
        auto s{0}, f{0};
        while(true){
            s = n[s];
            f = n[n[f]];
            if(s==f){
                break;
            }
        }
        int s2{0};
        while(true){
            s = n[s];
            s2 = n[s2];
            if(s==s2) return s;
        }
    }
};
